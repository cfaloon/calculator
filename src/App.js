import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';
import Display from './Display.js';
import Button from './Button.js';

class App extends Component {
  render() {
    let displayVal = 0;
    let buttonArray = [];
    for(let x=0; x <= 9; x++) {
      buttonArray.push(<Button key={x} value={x} className="Button-numeral" />);
    }


    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">Calculator</h1>
        </header>
        <div className="Interface col-md-offset-3 col-md-5">
          <div className="row">
            <div className="col-md-4">
              <Display inputValue={displayVal} />
            </div>
          </div>
          <div className="row">
            <div className="col-md-4">
              <div className="buttons">
                { buttonArray[7] }{ buttonArray[8] }{ buttonArray[9] }<Button value="*" className="Button-operator"/>
              </div>
            </div>
          </div>
          <div className="row">
            <div className="col-md-4">
              <div className="buttons">
                { buttonArray[4] }{ buttonArray[5] }{ buttonArray[6] }<Button value="/" className="Button-operator"/>
              </div>
            </div>
          </div>
          <div className="row">
            <div className="col-md-4">
              <div className="buttons">
                { buttonArray[1] }{ buttonArray[2] }{ buttonArray[3] }<Button value="+" className="Button-operator"/>
              </div>
            </div>
          </div>
          <div className="row">
            <div className="col-md-4">
              <div className="buttons">
                <Button value="C" className="Button-clear"/>{ buttonArray[0] }<Button value="=" className="Button-clear"/><Button value="/" className="Button-operator"/>
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

export default App;
