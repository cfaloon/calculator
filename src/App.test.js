import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
});

// SNAPSHOT TESTS follow link for more details
// http://facebook.github.io/jest/docs/en/snapshot-testing.html#snapshot-testing-with-jest
it('enters numbers to display', () => {
  const app = renderer.create(<App />);
  expect(app.toJSON()).toMatchSnapshot();

  app.getInstance().handleReceiveInput('4');
  expect(app.toJSON()).toMatchSnapshot();

  app.getInstance().handleReceiveInput('64');
  expect(app.toJSON()).toMatchSnapshot();
});

it('clears screen', () => {
  const app = renderer.create(<App />);
  expect(app.toJSON()).toMatchSnapshot();

  app.getInstance().handleReceiveInput('4');
  expect(app.toJSON()).toMatchSnapshot();

  app.getInstance().handleReceiveInput('0');
  expect(app.toJSON()).toMatchSnapshot();
});

it('performs addition', () => {
  const app = renderer.create(<App />);
  expect(app.toJSON()).toMatchSnapshot();

  app.getInstance().handleReceiveInput('4');
  expect(app.toJSON()).toMatchSnapshot();

  app.getInstance().handleReceiveOperator('+');
  expect(app.toJSON()).toMatchSnapshot();

  app.getInstance().handleReceiveInput('7');
  expect(app.toJSON()).toMatchSnapshot();

  app.getInstance().handleReceiveOperator('=');
  expect(app.toJSON()).toMatchSnapshot();
});
