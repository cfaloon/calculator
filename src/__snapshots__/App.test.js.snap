// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`clears screen 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <h1
      className="App-title"
    >
      Calculator
    </h1>
  </header>
  <div
    className="Interface"
  >
    <input
      className="display"
      id="display"
      readOnly={true}
      value={0}
    />
    <div
      className="button-row"
    >
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        7
      </button>
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        8
      </button>
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        9
      </button>
      <button
        className="Button-operator"
        onClick={[Function]}
      >
        *
      </button>
    </div>
    <div
      className="button-row"
    >
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        4
      </button>
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        5
      </button>
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        6
      </button>
      <button
        className="Button-operator"
        onClick={[Function]}
      >
        /
      </button>
    </div>
    <div
      className="button-row"
    >
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        1
      </button>
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        2
      </button>
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        3
      </button>
      <button
        className="Button-operator"
        onClick={[Function]}
      >
        +
      </button>
    </div>
    <div
      className="button-row"
    >
      <button
        className="Button-clear"
        onClick={[Function]}
      >
        C
      </button>
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        0
      </button>
      <button
        className="Button-operator"
        onClick={[Function]}
      >
        =
      </button>
      <button
        className="Button-operator"
        onClick={[Function]}
      >
        -
      </button>
    </div>
  </div>
</div>
`;

exports[`clears screen 2`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <h1
      className="App-title"
    >
      Calculator
    </h1>
  </header>
  <div
    className="Interface"
  >
    <input
      className="display"
      id="display"
      readOnly={true}
      value={4}
    />
    <div
      className="button-row"
    >
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        7
      </button>
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        8
      </button>
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        9
      </button>
      <button
        className="Button-operator"
        onClick={[Function]}
      >
        *
      </button>
    </div>
    <div
      className="button-row"
    >
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        4
      </button>
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        5
      </button>
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        6
      </button>
      <button
        className="Button-operator"
        onClick={[Function]}
      >
        /
      </button>
    </div>
    <div
      className="button-row"
    >
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        1
      </button>
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        2
      </button>
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        3
      </button>
      <button
        className="Button-operator"
        onClick={[Function]}
      >
        +
      </button>
    </div>
    <div
      className="button-row"
    >
      <button
        className="Button-clear"
        onClick={[Function]}
      >
        C
      </button>
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        0
      </button>
      <button
        className="Button-operator"
        onClick={[Function]}
      >
        =
      </button>
      <button
        className="Button-operator"
        onClick={[Function]}
      >
        -
      </button>
    </div>
  </div>
</div>
`;

exports[`clears screen 3`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <h1
      className="App-title"
    >
      Calculator
    </h1>
  </header>
  <div
    className="Interface"
  >
    <input
      className="display"
      id="display"
      readOnly={true}
      value={0}
    />
    <div
      className="button-row"
    >
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        7
      </button>
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        8
      </button>
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        9
      </button>
      <button
        className="Button-operator"
        onClick={[Function]}
      >
        *
      </button>
    </div>
    <div
      className="button-row"
    >
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        4
      </button>
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        5
      </button>
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        6
      </button>
      <button
        className="Button-operator"
        onClick={[Function]}
      >
        /
      </button>
    </div>
    <div
      className="button-row"
    >
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        1
      </button>
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        2
      </button>
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        3
      </button>
      <button
        className="Button-operator"
        onClick={[Function]}
      >
        +
      </button>
    </div>
    <div
      className="button-row"
    >
      <button
        className="Button-clear"
        onClick={[Function]}
      >
        C
      </button>
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        0
      </button>
      <button
        className="Button-operator"
        onClick={[Function]}
      >
        =
      </button>
      <button
        className="Button-operator"
        onClick={[Function]}
      >
        -
      </button>
    </div>
  </div>
</div>
`;

exports[`enters numbers to display 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <h1
      className="App-title"
    >
      Calculator
    </h1>
  </header>
  <div
    className="Interface"
  >
    <input
      className="display"
      id="display"
      readOnly={true}
      value={0}
    />
    <div
      className="button-row"
    >
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        7
      </button>
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        8
      </button>
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        9
      </button>
      <button
        className="Button-operator"
        onClick={[Function]}
      >
        *
      </button>
    </div>
    <div
      className="button-row"
    >
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        4
      </button>
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        5
      </button>
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        6
      </button>
      <button
        className="Button-operator"
        onClick={[Function]}
      >
        /
      </button>
    </div>
    <div
      className="button-row"
    >
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        1
      </button>
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        2
      </button>
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        3
      </button>
      <button
        className="Button-operator"
        onClick={[Function]}
      >
        +
      </button>
    </div>
    <div
      className="button-row"
    >
      <button
        className="Button-clear"
        onClick={[Function]}
      >
        C
      </button>
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        0
      </button>
      <button
        className="Button-operator"
        onClick={[Function]}
      >
        =
      </button>
      <button
        className="Button-operator"
        onClick={[Function]}
      >
        -
      </button>
    </div>
  </div>
</div>
`;

exports[`enters numbers to display 2`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <h1
      className="App-title"
    >
      Calculator
    </h1>
  </header>
  <div
    className="Interface"
  >
    <input
      className="display"
      id="display"
      readOnly={true}
      value={4}
    />
    <div
      className="button-row"
    >
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        7
      </button>
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        8
      </button>
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        9
      </button>
      <button
        className="Button-operator"
        onClick={[Function]}
      >
        *
      </button>
    </div>
    <div
      className="button-row"
    >
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        4
      </button>
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        5
      </button>
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        6
      </button>
      <button
        className="Button-operator"
        onClick={[Function]}
      >
        /
      </button>
    </div>
    <div
      className="button-row"
    >
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        1
      </button>
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        2
      </button>
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        3
      </button>
      <button
        className="Button-operator"
        onClick={[Function]}
      >
        +
      </button>
    </div>
    <div
      className="button-row"
    >
      <button
        className="Button-clear"
        onClick={[Function]}
      >
        C
      </button>
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        0
      </button>
      <button
        className="Button-operator"
        onClick={[Function]}
      >
        =
      </button>
      <button
        className="Button-operator"
        onClick={[Function]}
      >
        -
      </button>
    </div>
  </div>
</div>
`;

exports[`enters numbers to display 3`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <h1
      className="App-title"
    >
      Calculator
    </h1>
  </header>
  <div
    className="Interface"
  >
    <input
      className="display"
      id="display"
      readOnly={true}
      value={46}
    />
    <div
      className="button-row"
    >
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        7
      </button>
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        8
      </button>
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        9
      </button>
      <button
        className="Button-operator"
        onClick={[Function]}
      >
        *
      </button>
    </div>
    <div
      className="button-row"
    >
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        4
      </button>
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        5
      </button>
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        6
      </button>
      <button
        className="Button-operator"
        onClick={[Function]}
      >
        /
      </button>
    </div>
    <div
      className="button-row"
    >
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        1
      </button>
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        2
      </button>
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        3
      </button>
      <button
        className="Button-operator"
        onClick={[Function]}
      >
        +
      </button>
    </div>
    <div
      className="button-row"
    >
      <button
        className="Button-clear"
        onClick={[Function]}
      >
        C
      </button>
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        0
      </button>
      <button
        className="Button-operator"
        onClick={[Function]}
      >
        =
      </button>
      <button
        className="Button-operator"
        onClick={[Function]}
      >
        -
      </button>
    </div>
  </div>
</div>
`;

exports[`performs addition 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <h1
      className="App-title"
    >
      Calculator
    </h1>
  </header>
  <div
    className="Interface"
  >
    <input
      className="display"
      id="display"
      readOnly={true}
      value={0}
    />
    <div
      className="button-row"
    >
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        7
      </button>
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        8
      </button>
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        9
      </button>
      <button
        className="Button-operator"
        onClick={[Function]}
      >
        *
      </button>
    </div>
    <div
      className="button-row"
    >
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        4
      </button>
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        5
      </button>
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        6
      </button>
      <button
        className="Button-operator"
        onClick={[Function]}
      >
        /
      </button>
    </div>
    <div
      className="button-row"
    >
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        1
      </button>
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        2
      </button>
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        3
      </button>
      <button
        className="Button-operator"
        onClick={[Function]}
      >
        +
      </button>
    </div>
    <div
      className="button-row"
    >
      <button
        className="Button-clear"
        onClick={[Function]}
      >
        C
      </button>
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        0
      </button>
      <button
        className="Button-operator"
        onClick={[Function]}
      >
        =
      </button>
      <button
        className="Button-operator"
        onClick={[Function]}
      >
        -
      </button>
    </div>
  </div>
</div>
`;

exports[`performs addition 2`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <h1
      className="App-title"
    >
      Calculator
    </h1>
  </header>
  <div
    className="Interface"
  >
    <input
      className="display"
      id="display"
      readOnly={true}
      value={4}
    />
    <div
      className="button-row"
    >
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        7
      </button>
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        8
      </button>
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        9
      </button>
      <button
        className="Button-operator"
        onClick={[Function]}
      >
        *
      </button>
    </div>
    <div
      className="button-row"
    >
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        4
      </button>
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        5
      </button>
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        6
      </button>
      <button
        className="Button-operator"
        onClick={[Function]}
      >
        /
      </button>
    </div>
    <div
      className="button-row"
    >
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        1
      </button>
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        2
      </button>
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        3
      </button>
      <button
        className="Button-operator"
        onClick={[Function]}
      >
        +
      </button>
    </div>
    <div
      className="button-row"
    >
      <button
        className="Button-clear"
        onClick={[Function]}
      >
        C
      </button>
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        0
      </button>
      <button
        className="Button-operator"
        onClick={[Function]}
      >
        =
      </button>
      <button
        className="Button-operator"
        onClick={[Function]}
      >
        -
      </button>
    </div>
  </div>
</div>
`;

exports[`performs addition 3`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <h1
      className="App-title"
    >
      Calculator
    </h1>
  </header>
  <div
    className="Interface"
  >
    <input
      className="display"
      id="display"
      readOnly={true}
      value={0}
    />
    <div
      className="button-row"
    >
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        7
      </button>
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        8
      </button>
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        9
      </button>
      <button
        className="Button-operator"
        onClick={[Function]}
      >
        *
      </button>
    </div>
    <div
      className="button-row"
    >
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        4
      </button>
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        5
      </button>
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        6
      </button>
      <button
        className="Button-operator"
        onClick={[Function]}
      >
        /
      </button>
    </div>
    <div
      className="button-row"
    >
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        1
      </button>
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        2
      </button>
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        3
      </button>
      <button
        className="Button-operator"
        onClick={[Function]}
      >
        +
      </button>
    </div>
    <div
      className="button-row"
    >
      <button
        className="Button-clear"
        onClick={[Function]}
      >
        C
      </button>
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        0
      </button>
      <button
        className="Button-operator"
        onClick={[Function]}
      >
        =
      </button>
      <button
        className="Button-operator"
        onClick={[Function]}
      >
        -
      </button>
    </div>
  </div>
</div>
`;

exports[`performs addition 4`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <h1
      className="App-title"
    >
      Calculator
    </h1>
  </header>
  <div
    className="Interface"
  >
    <input
      className="display"
      id="display"
      readOnly={true}
      value={7}
    />
    <div
      className="button-row"
    >
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        7
      </button>
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        8
      </button>
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        9
      </button>
      <button
        className="Button-operator"
        onClick={[Function]}
      >
        *
      </button>
    </div>
    <div
      className="button-row"
    >
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        4
      </button>
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        5
      </button>
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        6
      </button>
      <button
        className="Button-operator"
        onClick={[Function]}
      >
        /
      </button>
    </div>
    <div
      className="button-row"
    >
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        1
      </button>
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        2
      </button>
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        3
      </button>
      <button
        className="Button-operator"
        onClick={[Function]}
      >
        +
      </button>
    </div>
    <div
      className="button-row"
    >
      <button
        className="Button-clear"
        onClick={[Function]}
      >
        C
      </button>
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        0
      </button>
      <button
        className="Button-operator"
        onClick={[Function]}
      >
        =
      </button>
      <button
        className="Button-operator"
        onClick={[Function]}
      >
        -
      </button>
    </div>
  </div>
</div>
`;

exports[`performs addition 5`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <h1
      className="App-title"
    >
      Calculator
    </h1>
  </header>
  <div
    className="Interface"
  >
    <input
      className="display"
      id="display"
      readOnly={true}
      value={11}
    />
    <div
      className="button-row"
    >
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        7
      </button>
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        8
      </button>
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        9
      </button>
      <button
        className="Button-operator"
        onClick={[Function]}
      >
        *
      </button>
    </div>
    <div
      className="button-row"
    >
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        4
      </button>
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        5
      </button>
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        6
      </button>
      <button
        className="Button-operator"
        onClick={[Function]}
      >
        /
      </button>
    </div>
    <div
      className="button-row"
    >
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        1
      </button>
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        2
      </button>
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        3
      </button>
      <button
        className="Button-operator"
        onClick={[Function]}
      >
        +
      </button>
    </div>
    <div
      className="button-row"
    >
      <button
        className="Button-clear"
        onClick={[Function]}
      >
        C
      </button>
      <button
        className="Button-numeral"
        onClick={[Function]}
      >
        0
      </button>
      <button
        className="Button-operator"
        onClick={[Function]}
      >
        =
      </button>
      <button
        className="Button-operator"
        onClick={[Function]}
      >
        -
      </button>
    </div>
  </div>
</div>
`;
